libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/auth/RustMatrixAuthenticationService.kt

```diff
@@ -353,7 +354,7 @@ class RustMatrixAuthenticationService @Inject constructor(
     }
 
     private fun clear() {
-        currentClient?.close()
+        Disposable.destroy(currentClient)
```

Comment:
There is maybe one thing, the `close()` method is annotated with `@Synchronized` and `destroy()` is not, but has an `AtomicBoolean` to protect from multiple execution, so I think this is fine. WDYT @jmartinesp ?

Reply:
I think it should be fine.